zebra/zebra_srv6_vty.c

```diff
@@ -0,0 +1,327 @@
+/*
+ * Zebra SRv6 VTY functions
+ * Copyright (C) 2020  Hiroki Shirokura, LINE Corporation
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; see the file COPYING; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include <zebra.h>
+
+#include "memory.h"
+#include "if.h"
+#include "prefix.h"
+#include "command.h"
+#include "table.h"
+#include "rib.h"
+#include "nexthop.h"
+#include "vrf.h"
+#include "srv6.h"
+#include "lib/json.h"
+
+#include "zebra/zserv.h"
+#include "zebra/zebra_vrf.h"
+#include "zebra/zebra_srv6.h"
+#include "zebra/zebra_srv6_vty.h"
+#include "zebra/zebra_rnh.h"
+#include "zebra/redistribute.h"
+#include "zebra/zebra_routemap.h"
+#include "zebra/zebra_dplane.h"
+
+
+static struct cmd_node sr_node = {
+	SR_NODE, "%s(config-sr)# ", 1
+};
+
+static struct cmd_node srv6_node = {
+	SRV6_NODE, "%s(config-srv6)# ", 1
+};
+
+static struct cmd_node srv6_locs_node = {
+	SRV6_LOCS_NODE, "%s(config-srv6-locators)# ", 1
+};
+
+static struct cmd_node srv6_loc_node = {
+	SRV6_LOC_NODE, "%s(config-srv6-locator)# ", 1
+};
+
+DEFUN (show_srv6_sid,
+       show_srv6_sid_cmd,
+       "show segment-routing srv6 sid [json]",
+       SHOW_STR
+       "Segment Routing\n"
+       "Segment Routing SRv6\n"
+       "SID Information\n"
+       JSON_STR)
+{
+	const bool uj = use_json(argc, argv);
+	struct route_node *rn;
+	struct route_table *table;
+	json_object *json = NULL;
+	json_object *json_sids = NULL;
+	json_object *json_sid = NULL;
+
+	if (uj) {
+		json = json_object_new_object();
+		json_sids = json_object_new_array();
+		json_object_object_add(json, "localSids", json_sids);
+	} else {
+		vty_out(vty, "Local SIDs:\n");
+		vty_out(vty, " Name       Context              Prefix                   Owner\n");
+		vty_out(vty, "---------- -------------------- ------------------------ ------------\n");
+	}
+
+	table = zebra_vrf_table(AFI_IP6, SAFI_UNICAST, VRF_DEFAULT);
+
+	for (rn = route_top(table);
+	     rn; rn = srcdest_route_next(rn)) {
+		struct route_entry *re;
+
+		RNODE_FOREACH_RE (rn, re) {
+			struct nexthop_group *nhg = re->nhe->nhg;
```

Comment:
While this PR is waiting, `struct nhg_hash_entry` was updated.
https://github.com/FRRouting/frr/pull/5857/files#diff-7902fbfbd4f206b974e7e7eb06a086d4

So, I need to fix here like-a following.
```diff
- struct nexthop_group *nhg = re->nhe->nhg;
+ struct nexthop_group *nhg = &re->nhe->nhg;
```

Reply:
And I need to rebase today's master branch.

Reply:
I fixed it. on this force-push

all-force-push: https://github.com/FRRouting/frr/compare/346f80d7b533c8edd3201884b5c4cb93d355c1fc..315a367ef64b1ec9fec53d3a1e7b4362e1471724

about-this-topic: 
https://github.com/FRRouting/frr/compare/346f80d7b533c8edd3201884b5c4cb93d355c1fc..315a367ef64b1ec9fec53d3a1e7b4362e1471724#diff-970ffcf82710594811447a0412ad5680